packages/gasket-utils/lib/lazy-load-package.js

```diff
@@ -0,0 +1,48 @@
+const fs = require('fs/promises');
+const path = require('path');
+const PackageManager = require('./package-manager');
+const tryResolve = require('./try-resolve');
+
+/**
+ * Determine package manager
+ * @param {string} root Gasket root
+ * @param {Logger} logger Gasket logger
+ * @returns {string} Package manager cmd
+ */
+async function getPkgManager(root, logger) {
+  const yarnLock = await fs.readFile(path.join(root, 'yarn.lock'), 'utf8').catch(() => {
+    logger.info('LazyLoadPackage - Installing using npm');
+  });
+
+  if (yarnLock) {
+    logger.info('LazyLoadPackage - Installing using yarn');
+    return 'yarn';
+  }
+
+  return 'npm';
+}
+
+/**
+ * Lazy load package - load devDependency programmatically when needed
+ * @param {string} dependency The require'ed dep needed
+ * @param {Gasket} gasket Gasket instance
+ * @returns {object} module
+ */
+module.exports = async function lazyLoadPackage(dependency, gasket) {
+  const { logger } = gasket;
+  const { root } = gasket.config;
+  const cmd = await getPkgManager(root, logger);
+  const pkg = dependency.split('/')[0];
```

Comment:
This is going to break with scoped packages, i.e. lazy loading
`@my/package/deep/export/module`
would need to install
`@my/package`

Reply:
With a regex like this:
`const rePackage = /^(@[\w.]+)?\/?([\w.]+)/;`

You could probably change this to
```suggestion
  const pkg = dependency.match(rePackage)[0];
```